File: SchoolWebApp-API/Project.Core/Entities/Staff/StaffAttendance.cs
using SchoolWebApp.Core.Entities.Shared;

namespace SchoolWebApp.Core.Entities.Staff
{
    public  class StaffAttendance: Attendance
    {
        public int StaffDetailsId { get; set; }
        public StaffDetails StaffDetails { get; set; }  
    }
}

File: SchoolWebApp-Client/src/app/staff/models/staff-attendance.ts
import {ResourceModel} from '@/core/models/ResourceModel';
import {StaffDetails} from './staff-details';

export class StaffAttendance extends ResourceModel<StaffAttendance> {
    public staffDetailsId?: number;
    public staffDetails?: StaffDetails;
    public date?: Date;
    public present?: boolean;
    public remarks?: string;

    public timeIn?: string;
    public timeOut?: string;
    
    constructor(model?: Partial<StaffAttendance>) {
        super(model);
    }
}
